src/ContextMenu.js

```diff
@@ -0,0 +1,126 @@
+const React = require('react');
+const ReactDOM = require('react-dom');
+const classNames = require('classnames');
+
+const Backdrop = require('./Backdrop');
+
+const MENU_RIGHT_SPACING = 300;
+const MENU_BOTTOM_SPACING = 160;
+
+/**
+ * Helper to display a context menu.
+ *
+ *  <ContextMenu component={() => <Dropdown></Dropdown>}>
+ *    Right click here !
+ *  </ContextMenu>
+ *
+ * @type {ReactClass}
+ */
+const ContextMenu = React.createClass({
+    propTypes: {
+        component:   React.PropTypes.func.isRequired,
+        children: React.PropTypes.node.isRequired
+    },
+
+    getInitialState() {
+        return {
+            open: false,
+            x: 0,
+            y: 0,
+            directionH: 'e'
+        };
+    },
+
+    /**
+     * When user is opening context menu.
+     */
+    onOpen(event) {
+        event.preventDefault();
+        event.stopPropagation();
+
+        const x = event.clientX;
+        const y = event.clientY;
+
+        const width = window.innerWidth;
+        const height = window.innerHeight;
+
+        this.setState({
+            open: true,
+            x,
+            y,
+            directionH: x > (width - MENU_RIGHT_SPACING) ? 'w' : 'e',
+            directionV: y > (height - MENU_BOTTOM_SPACING) ? 'n' : 's'
+        });
+    },
+
+    /**
+     * When user is closing the menu.
+     */
+    onClose() {
+        this.setState({
+            open: false
+        });
+    },
+
+    /**
+     * Close the menu after any click on the window.
+     * We wait next event loop to avoid hiding the menu before the click got propaged to it.
+     */
+    onWindowClick() {
+        setTimeout(() => this.onClose(), 1);
```

Comment:
I like that. Do you think we could use this trick for Dropdowns too (default configurable) ? To avoid having to clone elements like items, etc.